src/ape/types/__init__.py

```diff
@@ -197,15 +244,21 @@ def validate_address(cls, value):
 
         return convert(value, AddressType)
 
-    @property
+    @cached_property
```

Comment:
I think we have a lot of `@property` calls that should probably be `@cached_property`. A problem for another day